runelite-client/src/main/java/net/runelite/client/plugins/clancaller/ClanCallerPlugin.java

```diff
@@ -0,0 +1,160 @@
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.clancaller;
+
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Provides;
+import java.awt.Color;
+import java.util.ArrayList;
+import java.util.List;
+import javax.inject.Inject;
+import lombok.Getter;
+import net.runelite.api.Actor;
+import net.runelite.api.Client;
+import net.runelite.api.MenuEntry;
+import net.runelite.api.Player;
+import net.runelite.api.Query;
+import net.runelite.api.events.ConfigChanged;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.MenuOpenChanged;
+import net.runelite.api.queries.PlayerQuery;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.util.QueryRunner;
+
+@PluginDescriptor(
+	name = "Clan Caller"
+)
+public class ClanCallerPlugin extends Plugin
+{
+	private List<MenuEntry> menuEntries = new ArrayList<>();
+
+	@Getter
+	private Player caller;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private ClanCallerConfig config;
+
+	@Inject
+	private QueryRunner queryRunner;
+
+	@Inject
+	private ConfigManager configManager;
+
+	@Inject
+	private ClanCallerOverlay overlay;
+
+	@Provides
+	ClanCallerConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(ClanCallerConfig.class);
+	}
+
+	@Override
+	public Overlay getOverlay()
+	{
+		return overlay;
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (event.getKey().equals("getCaller"))
+		{
+			//reset caller when string changes
+			caller = null;
+			return;
+		}
+	}
+
+	@Subscribe
+	public void onMenuChanged(MenuOpenChanged event)
+	{
+		if (!client.isMenuOpen() ||
+			!config.recolorMenu() ||
+			caller == null ||
+			caller.getInteracting() == null)
+		{
+			return;
+		}
+
+		menuEntries.clear();
+
+		for (MenuEntry entry : client.getMenuEntries())
+		{
+			//Only change color on the Attack/Cast option
+			if (entry.getOption().equals("Attack") || entry.getOption().equals("Cast"))
+			{
+				//Names with a space are a non-breaking space
+				String entryTarget = entry.getTarget().replaceAll("\\u00a0", " ");
+				Actor callerTarget = caller.getInteracting();
+
+				if (entryTarget.contains(callerTarget.getName()))
```

Comment:
You should probably add back the colour tag to the the callerTarget and use equals method to avoid problems with names that contain other names (e.g. Zezima and Zezima123)

Reply:
The reason it is a contains method instead of equals is because the right click menu target for a name is actually "<col=ffffff>Zezima</col> <col=ff0000>(level-126)</col>". The problem you mentioned still needs to be fixed though so I will figure out a way to maybe split the string.